### PR TITLE
[WIP] Install from official repos

### PR DESCRIPTION
**Pull request by madAndroid**
Provides support for installing from official Elastic.co repos - addresses #14
[Work In Progress] Currently mostly untested - in the process of testing, will update with progress
- [x] Test on RedHat `$osfamily`
- [ ] Test on Debian `$osfamily`

### CODE REVIEW

**Comment by lesaux** [comment]
Hello, this looks great!
I will only be able to take a better look at it early next week.
Thanks for your work.

**Comment by madAndroid** [comment]
> Hello, this looks great!
> I will only be able to take a better look at it early next week.
> Thanks for your work.

No problem :) it seems to be working well so far .. there are a few concerns about the new packages from Elastic.co - the main one being the lack of logrotate config ... I've added that in a separate branch, along with config to set log paths, will create another PR to deal with that at some point

File: manifests/config.pp [hunk 1]
```diff
@@ -6,7 +6,13 @@
 
   $version = $kibana4::package_ensure
 
-  file { "${kibana4::install_dir}/kibana-${version}/config/kibana.yml":
+  if $kibana4::use_official_repos {
```
Comment (eperdeme) [inline manifests/config.pp:9]: Can you make this use the if package_provide = 'package'. We use the official repo but a mirror as no external access and this PR hard sets the mirrors to use.

**Comment by eperdeme** [comment]
I think you need to maybe modify the code to make better use of the vendors kibana package. The package comes with it's own init script and service user, so maybe if type = 'package' should stop the user/init script being clobbered by this module.

**Comment by antaflos** [comment]
I think this needs to cover three cases:
- Install Kibana from the tarball, like it is currently implemented
- Install Kibana from OS packages
  - In this case don't manage the init script
- _Optionally_ manage the official APT/RPM repo

The last one is critical because different Puppet infrastructures manage repositories in different manners. For example we use profiles that tie together the module, user/group creation, repo management, ordering dependencies and other related stuff. I would never want a module to manage a repository or user for me, because we do that in our own specialised modules.

For reference, our Kibana4 profile looks like this:

``` puppet
class profile::kibana4 {
  contain '::kibana4'

  realize (
    Account::Systemgroup['kibana4'],
    Account::Systemuser['kibana4'],
    Apt::Source['kibana4']
  )

  Account::Systemuser['kibana4']
  -> Apt::Source['kibana4']
  -> Class['::kibana4']
}
```

So I wouldn't want the kibana4 module to manage repos or users by default, because we do that in the profile beforehand. Of course it should be an option that can be enabled if the user wants it.

**Comment by lesaux** [comment]
Thanks for the hard work @madAndroid.
I've rebased your file changes on a previous merge, and made sure init.d script was not managed when installation from OS packages was chosen in #22

**Comment by lesaux** [comment]
Thanks for the hard work @madAndroid.
I've rebased your commit on a previous merge, and made sure init.d script was not managed when installation from OS packages was chosen.

**Comment by madAndroid** [comment]
@lesaux - Awesome, thanks for sorting that :) Any idea when we can expect a new release with these changes? Should we close this PR in favour of #22 ?

**Comment by lesaux** [comment]
@madAndroid . Yup I guess we can close this PR. I'll merge the changes in a few moments, and will publish a new package on the forge right after that. Thanks a lot for your help on this.

**Pull request closed without merging**